### ISSUE TITLE
Hi Rob,

### ISSUE THREAD

**Issue by broberts2** [Rob--/memoryjs#99]
Hi Rob,

I apologize in advance for commenting on an already closed issue, but I've been at this for hours and have hit a hard wall. I was able to read from cheat engine the values I want at a particular address just fine, but I'm having issues with the pointers in my code. As you can see in the GE screenshot, I am able to get float I want, but in my console screenshot from node, I am getting something totally different. _**pointerChain**_ is the method in question that is being called from another file. Any help would be greatly appreciated :)

Please see below for reference:

![image](https://user-images.githubusercontent.com/32076116/184535085-2782ad2b-16fd-4894-9ba8-eda5a0368591.png)
![image](https://user-images.githubusercontent.com/32076116/184535513-579ff9d7-080f-4086-863d-bb604e1b97b2.png)
![image](https://user-images.githubusercontent.com/32076116/184535163-b89bc5de-d383-40d8-95e4-6aebe887a3db.png)
![image](https://user-images.githubusercontent.com/32076116/184535179-2d834c43-b04c-4888-bf48-f4d2bce489b4.png)

_Originally posted by @broberts2 in https://github.com/Rob--/memoryjs/issues/89#issuecomment-1214350932_

**Comment by aurelius88** [comment]
The problem here is at **line 7**. You are just inserting an address offset (to the process' address), but `memoryjs.readMemory` only accepts the full address. 
So you need to add the process' base address to your offset `0x073dd720`. You can get the process' base address by using your `processObject`.
In the end, line 5 to 9 should look like this:
```js
const p1 = memoryjs.readMemory(
    processObject.handle,
    processObject.modBaseAddr + 0x073dd720,
    memoryjs.PTR
);
```

**Comment by broberts2** [comment]
Thanks for the reply,

I've tried that as well, and when I do that I still receive a separate value noted in the below screenshot:

![image](https://user-images.githubusercontent.com/32076116/185023277-a72a4268-254d-4ace-aef0-0c978f534a87.png)

**Comment by broberts2** [comment]
Here's some context if it helps.

That pointer is tracking the z player coordinate in-game. It is generally a float between 0.0- 3000.0 when standing above water and < 0.0 when diving in Sea of Thieves. When working correctly, that value updates accordingly each time I scan memory every half-second.

**Comment by Rob--** [comment]
Are you targeting a 32 or 64 bit application? And is your Node 32 or 64 bit?

It would help if you printed all the pointer values in hex to make it easier to see what is going wrong and where!

**Comment by broberts2** [comment]
Using Node 64 and also targeting a 64 bit app. I've uploaded several more screen captures after adding some more logging like you suggested. You're probably on to something with pointer 6:

![Capture](https://user-images.githubusercontent.com/32076116/185189516-20ce153e-36db-4d11-8017-9f7c3927be89.PNG)
![image](https://user-images.githubusercontent.com/32076116/185189461-095b86cd-93e2-418d-ada0-186758e2f1de.png)
![image](https://user-images.githubusercontent.com/32076116/185189919-1dbb8a37-8cc2-4c0c-8a09-8be583d2ac74.png)

**Comment by Rob--** [comment]
Checking with Cheat Engine like you did before, is the first pointer that is read the correct value?

**Comment by broberts2** [comment]
Interesting. So yeah, they all match up except for the last one:

![image](https://user-images.githubusercontent.com/32076116/185204052-a077c6a4-d954-48f2-afeb-75b176727a61.png)
![image](https://user-images.githubusercontent.com/32076116/185204117-1bb261e7-a013-433c-84b4-10a188844fc5.png)

**Comment by broberts2** [comment]
In fact, when running my node program, that last pointer address is constantly changing.

**Comment by broberts2** [comment]
Okay, so I actually figured this out. This was totally on my end. I included a snippet of my working code with the change.

This was really good to help me understand some fundamental concepts of pointers and I really appreciate you asking the right questions to make me think about the problem the correct way.

Thanks for the help Rob!

![image](https://user-images.githubusercontent.com/32076116/185205381-1bbcda03-93b3-458c-b8bc-64082989a63b.png)
